include/jwt-cpp/string_types.h

```diff
@@ -0,0 +1,24 @@
+#ifndef STRING_TYPES_H
+#define STRING_TYPES_H
+
+namespace jwt {
+#if __cplusplus >= 201703L
+
+#include <string_view>
+
+#define JWT_HAS_STRING_VIEW
+
+	using string_constant = std::string_view;
+	using string_view = std::string_view;
+
+#else
+
+#include <string>
+
+	using string_constant = std::string;
```

Comment:
This name seems misleading 🤔 a non-const constant string?

Reply:
The name is probably not the best one. It should only be used for definition of compile time constant strings. In C++17 we can (and it's the best solution) use `std::string_view`, but older C++ versions don't have it so the best we can do is to default on `std::string`.

These types are the building blocks of this PR, and makes it possible for pre C++17 compilers to still compile the library. Do you have a better name in mind ? For instance, `static_string` ?